addons/ingress-nginx/addon.rb

```diff
@@ -48,4 +50,8 @@ def default_backend_replicas
     return 2 if r < 2
     r
   end
+
+  def secret_name
```

Comment:
I don't think you need a specific method for this. As the attribute is marked as optional in the schema, if it's not provided it will be `nil` anyway. Without this method, you could use the value as `config.default_backend&.dig(:secret_name)` in the `apply_resources` vars